game-core/src/main/java/games/strategy/triplea/attachments/UnitAttachment.java

```diff
@@ -3607,6 +3591,19 @@ public String toStringShortAndOnlyImportantDifferences(final GamePlayer player)
     return result.toString();
   }
 
+  private void addLabeledUnitTypes(
```

Comment:
What are your thoughts to have this method return a `Tuple` rather than use side-effects?

https://github.com/triplea-game/triplea/wiki/Java-Code-Conventions#do-not-mutate-parameters-avoid-out-and-avoid-inout-parameters-5489

Reply:
It would duplicate more code because we'd need to move the if statement out of the function - either on the return value, by making the function return a null or optional tuple, or on the input param - which would then require an extra local if we don't want to recompute it twice.

So I prefer keeping it as is. Plus there's other functions already following this pattern.

Reply:
The indirection is problematic as well; side-effects and no-op methods that are 200 lines below where you are looking can make things also difficult to follow. Notably to understand what those two lines above do you need to head down the page and then back up.

We've the code convention established to declare essentially that existing examples are anti-patterns and not something we want to build on or continue.

Be what it may, there are  a number of problems here, I'd agree there are trade-offs to be made.